driverapi/driverapi.go

```diff
@@ -34,32 +34,91 @@ type Driver interface {
 	DeleteNetwork(nid types.UUID) error
 
 	// CreateEndpoint invokes the driver method to create an endpoint
-	// passing the network id, endpoint id and driver
-	// specific config. The config mechanism will eventually be replaced
-	// with labels which are yet to be introduced.
-	CreateEndpoint(nid, eid types.UUID, options map[string]interface{}) (*sandbox.Info, error)
+	// passing the network id, endpoint id endpoint information and driver
+	// specific config. The endpoint information can be either consumed by
+	// the driver or populated by the driver. The config mechanism will
+	// eventually be replaced with labels which are yet to be introduced.
+	CreateEndpoint(nid, eid types.UUID, epInfo EndpointInfo, options map[string]interface{}) error
 
 	// DeleteEndpoint invokes the driver method to delete an endpoint
 	// passing the network id and endpoint id.
 	DeleteEndpoint(nid, eid types.UUID) error
 
 	// EndpointInfo retrieves from the driver the operational data related to the specified endpoint
-	EndpointInfo(nid, eid types.UUID) (map[string]interface{}, error)
+	EndpointOperInfo(nid, eid types.UUID) (map[string]interface{}, error)
```

Comment:
Being an external Facing API, it would be okay to be verbose : `EndpointOperationalInfo`